content/languages/javascript/authoring.md

```diff
@@ -0,0 +1,281 @@
+
+
+## General info
+
+Any technical information related to the JavaScript setup on Codewars can be found on the [JavaScript reference](/languages/javascript/) page (language versions, available libraries, and setup of the code runner).
+
+## Description
+
+JavaScript code blocks can be inserted with JavaScript-specific part in [sequential code blocks](/references/markdown/extensions/#sequential-code-blocks):
+
+~~~
+```javascript
+
+...your code here...
+
+```
+~~~
+
+JavaScript-specific paragraphs can be inserted with [language conditional rendering](/references/markdown/extensions/#conditional-rendering):
+
+```
+~~~if:javascript
+
+...text visible only for JavaScript description...
+
+~~~
+
+~~~if-not:javascript
+
+...text not visible in JavaScript description...
+
+~~~
+```
+
+## Tasks and Requirements
+
+Some concepts don't always translate well to or from JavaScript. Because of this, some constructs should be avoided and some translations just shouldn't be done.
+- Avoid returning different data types depending on the situation (_"Return the result, or the string 'Error' if no result can be found..."_). JavaScript is dynamically typed, which is not the case for some other languages. Returning `null` might be appropriate in some situations, but throwing an error might be better in others.
+
+Some kata just should not be translated into JavaScript because it can be difficult to keep their initial idea:
+- Some kata might be meant for another language specifically. For example, a kata about learning to use Python's `itertools.takewhile` function to do something should not be translated to JavaScript.
+- The Codewars runner provides a set of preinstalled packages that are available for users solving the kata. They can be a real game-changer when it comes to the difficulty of a JavaScript translation. For example, `lodash` or `ramda` include many helpful utilities, which could make some kata trivial that wouldn't be trivial in other languages.
+
+## Tests
+
+### Mocha
+
+JavaScript kata should use [Mocha](https://mochajs.org/) to implement and execute tests.
+
+#### Test grouping functions
+
+To create and present test output, Mocha uses parameters of the `describe` and `it` functions:
+
+```javascript
+describe('Fixed tests', () => {
+  it('Odd numbers', () => {
+    // some assertions...
+  });
+
+  it('Even numbers', () => {
+    // some assertions...
+  });
+});
+
+describe('Random tests', () => {
+  it('Small inputs', () => {
+    // some assertions...
+  });
+
+  it('Large inputs', () => {
+    // some assertions...
+  });
+});
+```
+
+`describe` blocks can contain `describe` or `it` blocks, but `it` blocks can only contain assertions.
+
+#### Dynamically generated test cases
+
+It's possible to put functions decorated with `it` in a loop and use them as a construct similar to parameterized test cases known from other testing frameworks, for example:
```

Comment:
I think it's jsut a copy/paste mistake, but test functions in Mocha are not really decorated with anything.

```suggestion
It's possible to put calls to `it` in a loop and use them as a construct similar to parameterized test cases known from other testing frameworks, for example:
```